pixtrack/utils/pytorch3d_render_utils.py

```diff
@@ -0,0 +1,122 @@
+import torch
+import trimesh
+import numpy as np
+import pycolmap
+from scipy.spatial.transform import Rotation as R
+from hloc.utils.read_write_model import Camera, Image
+from pytorch3d.io import load_objs_as_meshes, load_obj
+from pytorch3d.structures import Meshes
+from pytorch3d.vis.plotly_vis import AxisArgs, plot_batch_individually, plot_scene
+from pytorch3d.vis.texture_vis import texturesuv_image_matplotlib
+from pytorch3d.renderer import (
+    look_at_view_transform,
+    look_at_rotation,
+    FoVPerspectiveCameras,
+    PerspectiveCameras,
+    PointLights, 
+    AmbientLights,
+    DirectionalLights, 
+    Materials, 
+    RasterizationSettings, 
+    MeshRenderer, 
+    MeshRasterizer,  
+    SoftPhongShader,
+    TexturesUV,
+    TexturesVertex)
+
+
+def create_look_at_camera_poses(radius, subdivisions=2):
+    sph = trimesh.creation.icosphere(subdivisions=subdivisions, 
+                                     radius=radius, 
+                                     color=None)
+    v = sph.vertices
+    Rs = [look_at_rotation(torch.tensor(v[i]).unsqueeze(0).float()).squeeze().numpy() for i in range(v.shape[0])]
+    Rs = [R.from_matrix(x).as_matrix() for x in Rs]
+    Ts = [-Rs[i].T @ v[i] for i in range(v.shape[0])]
+    Rs = torch.tensor(np.stack(Rs))
+    Ts = torch.tensor(np.stack(Ts))
+    return Rs, Ts
+
+def create_look_at_poses_for_mesh(mesh_path, subdivisions=2):
+    if torch.cuda.is_available():
+        device = torch.device("cuda:0")
```

Comment:
make this a param? We can expose it later

Reply:
Done